# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/mine/hotfix-net-install.patch
# Copyright (C) 2006 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Of course we do not want curl in one place and wget at the other, also
there is no wget in the second stange anymore ...

  - Rene Rebe <rene@exactcode.de>

diff -ur mine-0.23/gasgui.c mine-0.23-fixed/gasgui.c
--- mine-0.23/gasgui.c	2006-06-30 18:20:36.000000000 +0200
+++ mine-0.23-fixed/gasgui.c	2006-06-30 20:02:47.000000000 +0200
@@ -116,14 +116,14 @@
 		snprintf(gemfile, 1024, "%s/%s-%s.%s",
 		         targetdir, p->name, p->version, sourcefmt);
 
-		snprintf(command, 1024, "wget -O '%s' -q '%s/%s/pkgs/%s-%s.%s'",
-		         gemfile, sourcedir, config, p->name, p->version, sourcefmt);
+		snprintf(command, 1024, "fget '%s/%s/pkgs/%s-%s.%s' > '%s'",
+		         sourcedir, config, p->name, p->version, sourcefmt, gemfile);
 		system(command);
 		snprintf(command, 1024, "test -s '%s' || rm -f %s", gemfile, gemfile);
 		system(command);
 		if ( !access(gemfile, F_OK) ) return gemfile;
 
-		snprintf(command, 1024, "wget -O '%s' -q '%s/%s/pkgs/%s.%s'",
-		         gemfile, sourcedir, config, p->name, sourcefmt);
+		snprintf(command, 1024, "fget '%s/%s/pkgs/%s.%s' > '%s'",
+		         sourcedir, config, p->name, sourcefmt, gemfile);
 		system(command);
 		snprintf(command, 1024, "test -s '%s' || rm -f %s", gemfile, gemfile);
diff -ur mine-0.23/readdb.c mine-0.23-fixed/readdb.c
--- mine-0.23/readdb.c	2006-06-30 18:20:36.000000000 +0200
+++ mine-0.23-fixed/readdb.c	2006-06-30 18:10:59.000000000 +0200
@@ -441,7 +441,7 @@
 		snprintf(filename, PATH_MAX, "gzip -d < %s/%s/pkgs/packages.db",
 		         sourcedir, config);
 	else
-		snprintf(filename, PATH_MAX, "wget -O - -nv %s/%s/pkgs/packages.db | gzip -d",
+		snprintf(filename, PATH_MAX, "fget %s/%s/pkgs/packages.db | gzip -d",
 		         sourcedir, config);
 	if ( (f = popen(filename, "r")) != NULL ) {
 		char *package, line[160];
--- mine-0.23/mine.c.vanilla	2024-04-13 18:30:01.409243650 +0200
+++ mine-0.23/mine.c	2024-04-13 18:33:02.665242855 +0200
@@ -177,17 +177,17 @@
 			if ( gemfile[c] == '/' ) gemfile[c] = '_';
 		if ( access(gemfile, F_OK) ) {
 			snprintf(command, 8192, "mkdir -p '%s';"
-				"curl -s -S %s -o '%s.part' '%s' && "
+				"fget '%s' > '%s.part' '%s' && "
 				"mv '%s.part' '%s'", GEMCACHE,
 				access(MINECURLOPT, F_OK) ? "" : MINECURLOPT,
-				gemfile, url, gemfile, gemfile);
+				url, gemfile, gemfile, gemfile);
 			if ( mine_mode_verbose )
-				printf("Downloading %s ..\n", gemfile);
+				printf("Downloading %s ...\n", gemfile);
 			system(command);
 		}
 		else
 			if ( mine_mode_verbose )
-				printf("Using cached %s ..\n", gemfile);
+				printf("Using cached %s ...\n", gemfile);
 	} else {
 		snprintf(gemfile, 4096, "%s", url);
 	}
